src/components/icon/_hamburger.less

```diff
@@ -0,0 +1,68 @@
+.hamburger-btn {
+  position: relative;
+  width: 30px;
+  height: 20px;
+  background: transparent;
+  border: none;
+  border-style: solid;
+  border-width: 4px 0;
+  color: @black;
+  border-color: @black;
+  transition: all 0.2s ease;
+
+  &:before,
+  &:after {
+    content: "";
+    display: block;
+    width: 100%;
+    height: 4px;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    background: currentColor;
+    transform: translate(-50%, -50%);
+    transition: transform 0.2s ease;
+  }
+}
+
+.hamburger-btn:hover {
```

Comment:
nest inside `.hamburger-btn` in the beginning.

Reply:
`.hamburger-btn` is a parent class and other classes like `.hamburger-x-btn`, `.hamburger-min-btn`, `.hamburger-larr-btn` determine what icon the button will transform into when toggled. Also, could you tell me what issue you're finding with the transition? When I test it locally, it seems fine to me (there are 2 lines, :before and :after, hidden in the middle bar of the hamburger icon. They both turn 45 degrees to form the 'X' while the top and bottom bars fade away)
![hamburger](https://user-images.githubusercontent.com/52718953/95796689-841f6600-0cbb-11eb-9a73-4b63d545684c.gif)

Reply:
I understand but I think the most used is hamburger to x so let's keep it simple by just adding those two states. By just adding `.active` to `.hamburger-btn` it should change to the x. As for the transition I think it looks good so forget my comment about it.